src/main/preference.js

```diff
@@ -22,19 +23,19 @@ class Preference {
   }
 
```

Comment:
Maybe a `checkNeedUpdate` method is needed.

Reply:
What did you mean? A update is needed if `defaultSettings.keyNames != userSettings.keyNames`.

Reply:
My opinion is that first check if it is needed to update in Preference constructor.

If it is needed, then update the preferences.md in `userDataPath`.

Reply:
Yes, this is done by `this.requiresUpdate = !hasSameKeys(defaultSettings, userSettings)`.